### PR TITLE
Setup Husky v 7 , and configured husky and lint-staged

### PR DESCRIPTION
**Pull request by chirayumit001**
@K-Kumar-01 bhaiya plz review this PR. 
Had some issues with previous PR as husky was not working properly after resolving conflicts hence worked on it hence new PR.

### CODE REVIEW

**Comment by K-Kumar-01** [comment]
@chirayumit001 
Why are there so many commits named testing husky?

**Comment by chirayumit001** [comment]
> @chirayumit001 Why are there so many commits named testing husky?

I was just configuring the husky. I should have chosen better name for the commit messages, will keep that in mind. I just used upper arrow key in my terminal and pressed enter thats why they have same messages.

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npx --no-install lint-staged
+npx lint-staged
```
Comment (K-Kumar-01) [inline .husky/pre-commit]: What's the difference between these two?

**Review by K-Kumar-01 — COMMENTED** [review]
Just a doubt and then can merge it.

File: .husky/pre-commit (hunk 1 above)
Comment (chirayumit001) [reply .husky/pre-commit]: Sorry bhaiiya I don't really know, followed the steps, I also thought about **npx --no-install lint-staged**, but thought might still add this using the command -> **npx husky add .husky/pre-commit "npx --no-install lint-staged"** as well, in order to follow the steps completely and make sure things worked this time.

Comment (K-Kumar-01) [reply .husky/pre-commit]: @chirayumit001 
Do one thing.  Try removing anyone and see what happens. If the result is same, revert it back. Do this for both. Once done, let me know. We can then go ahead with the merge.

Comment (K-Kumar-01) [reply .husky/pre-commit]: See [this](https://javascript.plainenglish.io/yes-its-npx-not-npm-the-difference-explained-58cbb202ec33) for the difference above.

Comment (chirayumit001) [reply .husky/pre-commit]: > @chirayumit001 Do one thing. Try removing anyone and see what happens. If the result is same, revert it back. Do this for both. Once done, let me know. We can then go ahead with the merge.

Done bhaiya, the result is same in both cases, it is working with either of them.

Thank you for the article as well bhaiya, cleared some confusions 🙌✨

Comment (K-Kumar-01) [reply .husky/pre-commit]: @chirayumit001 
Just try one more thing. Try changing the command to `npm run lint-staged`. If it works, nice. If not, do the below.
Just commit the changes for --no-install one and remove the other.

Final change and then we can merge.

Comment (chirayumit001) [reply .husky/pre-commit]: @K-Kumar-01 
npm run command failed bhaiya, added --no--install command and removed other one as instructed

**Pull request merged**